lib/runtime/README.md

```diff
@@ -83,7 +84,8 @@ fn main() -> error::Result<()> {
     let instance = instantiate(WASM, &import_object)?;
 
     let values = instance
-        .dyn_func("add_one")?
+        .exports
+        .get::<DynFunc>("add_one")?
```

Comment:
Somehow I feel that `DynFunc` and `Func` should be merged together, as they reference to the same thing, just with a different API for calling the function.

We can have something like:

```rust
    let add_one = instance
        .exports
        .get::<Func<i32, i32>>("add_one")?;
    // For dynfunc usage
    let values = add_one.call(&[Value::I32(42)])?;
    // For normal func usage
    let values = add_one.call_native(42)?;
```

Thoughts?

Reply:
That's a pretty interesting idea -- the problem is that this seems to remove a lot of the usefulness of DynFunc -- a big benefit of it is that it does not have a type signature which means you can just have `Vec<DynFunc>` and that you can operate on it without knowing or caring what the type signature is.  In the above example, we'd still need to know the type signature statically to be able to get it.

This sounds like something that will take more design work!

Reply:
To clarify, it may be possible to implement this in a way to get that effect easily but I don't see any obvious solutions right away

Reply:
Sounds good. Let's handle that apart.